src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/ActivatorUtilitiesTests.cs

```diff
@@ -0,0 +1,208 @@
+
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Microsoft.Extensions.DependencyInjection.Tests
+{
+    public class ActivatorUtilitiesTests
+    {
+        [Fact]
+        public void ShouldUseLongestAvailableConstructorOnlyIfConstructorsHaveTheSamePriority()
+        {
+            var services = new ServiceCollection();
+            services.AddScoped<B>();
+            services.AddScoped<S>();
+            using var provider = services.BuildServiceProvider();
+            var a = new A();
+            var c = new C();
+
+            var instance = ActivatorUtilities.CreateInstance<Creatable>(provider, a, c);
+
+            Assert.Null(instance.B);
+        }
+
+        [Theory]
+        [InlineData(typeof(DefaultConstructorFirst))]
+        [InlineData(typeof(DefaultConstructorLast))]
+        public void ChoosesDefaultConstructorNoMatterOrder(Type instanceType)
+        {
+            var services = new ServiceCollection();
+            using var provider = services.BuildServiceProvider();
+
+            var instance = ActivatorUtilities.CreateInstance(provider, instanceType);
+
+            Assert.NotNull(instance);
+        }
+
+        [Fact]
+        public void ShouldFixIssue_46132()
+        {
+            var services = new ServiceCollection();
+            services.AddScoped<S>();
+            using var provider = services.BuildServiceProvider();
+            var a = new A();
+            var c = new C();
+
+            var instance = ActivatorUtilities.CreateInstance<Creatable>(
+                provider, c, a);
+
+            Assert.NotNull(instance);
+            Assert.Same(a, instance.A);
+            Assert.Same(c, instance.C);
+            Assert.NotNull(instance.S);
+            Assert.Null(instance.B);
+        }
+
+        [Theory]
+        [MemberData(nameof(ActivatorUtilitiesData))]
+        public void ShouldFixIssue_42339(Type instanceType)
+        {
+            var data = new Dictionary<string, object>();
+            var serviceProvider = new FakeServiceProvider(t =>
+            {
+                if (t == typeof(FakeValidationStatus)) return FakeValidationStatus.Invalid;
+                throw new NotImplementedException();
+            });
+
+            var instance = (IFakeValidationResult)ActivatorUtilities
+                .CreateInstance(serviceProvider, instanceType, "description", data);
+
+            Assert.Equal(FakeValidationStatus.Invalid, instance.Status);
+        }
+
+        public static TheoryData<Type> ActivatorUtilitiesData
+        {
+            get => new TheoryData<Type>
+            {
+                typeof(FakeValidationResult),
+                typeof(FakeValidationResultOps)
+            };
+        }
+    }
+
+    internal class A { }
+    internal class B { }
+    internal class C { }
+    internal class S { }
+
+    internal class Creatable
+    {
+        public A A { get; }
+        public B? B { get; }
```

Comment:
```suggestion
        public B B { get; }
```

We don't have nullable annotations in our tests.